PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -552,25 +577,22 @@ struct LambdaCorrTableProducer {
 
     // get ccdb object
     TObject* obj = reinterpret_cast<TObject*>(ccdbObj->FindObject(Form("%s", vCorrFactStrings[cCorrFactHistType][part].c_str())));
-
-    if (obj->InheritsFrom("TH1F")) {
-      TH1F* hist = reinterpret_cast<TH1F*>(obj->Clone());
-      int ptBin = hist->GetXaxis()->FindBin(v0.pt());
-      return hist->GetBinContent(ptBin);
-    } else if (obj->InheritsFrom("TH2F")) {
-      TH2F* hist = reinterpret_cast<TH2F*>(obj->Clone());
-      int ptBin = hist->GetXaxis()->FindBin(v0.pt());
-      int rapBin = hist->GetYaxis()->FindBin(v0.yLambda());
-      return hist->GetBinContent(ptBin, rapBin);
-    } else if (obj->InheritsFrom("TH3F")) {
-      TH3F* hist = reinterpret_cast<TH3F*>(obj->Clone());
-      int ptBin = hist->GetXaxis()->FindBin(v0.pt());
-      int rapBin = hist->GetYaxis()->FindBin(v0.yLambda());
-      int vzBin = hist->GetZaxis()->FindBin(col.posZ());
-      return hist->GetBinContent(ptBin, rapBin, vzBin);
+    TH1F* hist = reinterpret_cast<TH1F*>(obj->Clone());
+    float retVal = 0.;
+
+    if (std::string(obj->ClassName()) == "TH1F") {
+      retVal = hist->GetBinContent(hist->FindBin(v0.pt()));
+    } else if (std::string(obj->ClassName()) == "TH2F") {
+      retVal = hist->GetBinContent(hist->FindBin(v0.pt(), v0.yLambda()));
+    } else if (std::string(obj->ClassName()) == "TH3F") {
+      retVal = hist->GetBinContent(hist->FindBin(v0.pt(), v0.yLambda(), col.posZ()));
```

Comment:
As said in the previous review, I would suggest using a `switch` over the number of dimensions of the histogram
In here, per particle, apart of constructing the three `string` objects, there are up to three string comparisons which have to be done char by char